### PR TITLE
Fix the queue when reprioritize

### PR DESCRIPTION
**Pull request by scw00**
This pr tries to fix the #2323.

### CODE REVIEW

**Comment by zwoop** [comment]
Testing on Docs now.

File: proxy/http2/Http2DependencyTree.h [hunk 1]
```diff
@@ -257,17 +266,31 @@ void
 Http2DependencyTree<T>::_change_parent(Node *node, Node *new_parent, bool exclusive)
 {
   node->parent->children.remove(node);
+  if (node->queued) {
+    node->parent->queue->erase(node->entry);
```
Comment (masaori335) [inline proxy/http2/Http2DependencyTree.h]: If the `queue` of parent become empty by this operation, the `queued` flag of parent should also be `false`.
We should do same operations to parent of parent recursively.
So I think we should walk up the tree to correct `queue` and `queued` flag of parents here. WDYT?

File: proxy/http2/test_Http2DependencyTree.cc [hunk 2]
```diff
@@ -498,6 +498,129 @@ REGRESSION_TEST(Http2DependencyTree_remove_2)(RegressionTest *t, int /* atype AT
...
+
+  Tree::Node *node_a = tree->find(1);
+  Tree::Node *node_c = tree->find(5);
+  Tree::Node *node_f = tree->find(11);
+
+  tree->activate(node_f);
+  tree->reprioritize(1, 7, true);
+
+  box.check(node_a->queue->in(node_f->entry), "F should be in A's queue");
+  // do we need to reset C's queued flag since C's queue is empty ??
```
Comment (masaori335) [inline proxy/http2/test_Http2DependencyTree.cc]: I think so. In case for C is reprioritized. IMO, walking up the tree in `_change_parent` will fix this.

**Comment by scw00** [comment]
I'll updated after Docs Test.
If it doesn't work, I think logging the `http2_con` is a better choice for us to address this case. Since I can not reproduce on my local machine .

**Comment by zwoop** [comment]
Still tripping up on ASAN.  :-/

**Comment by scw00** [comment]
Hi @masaori335 Can you please review again !

**Comment by zwoop** [comment]
If this still doesn't fix the ASAN issue, do we still want this for 7.1.1 ?

**Comment by scw00** [comment]
May be we need to block this pr until I address or reproduce this case.

**Comment by zwoop** [comment]
I don't know how to reproduce it, but I was browsing docs.trafficserver from Safari and Chrome at the same time, and it crashed. Not sure if it was me, or someone else though :-).

**Comment by zwoop** [comment]
I marked this as WIP for now, so we don't accidentally merge it.

**Comment by scw00** [comment]
👍

**Comment by scw00** [comment]
Hi @zwoop  @masaori335 Updated ! Can you please review again !!

**Comment by scw00** [comment]
Only fixed the conflicts !

**Comment by scw00** [comment]
@zwoop I think this pr is ready to launch ! Can you please backport to 7.1.x

**Comment by zwoop** [comment]
Can we hold off on this one to 7.1.2 ? Or do we need it for 7.1.1 ? I'm just about ready to make the 7.1.1 RC, and this has not been tested particularly well.

**Pull request merged**